inbox/mailsync/service.py

```diff
@@ -48,6 +50,24 @@ def shared_sync_event_queue_for_zone(zone):
     return SHARED_SYNC_EVENT_QUEUE_ZONE_MAP[queue_name]
 
 
+@cache
+def get_monitor_classes() -> dict[str, Type[BaseMailSyncMonitor]]:
+    """
+    Return a dictionary mapping provider names to their respective monitor
+    """
+    monitor_classes = {
+        module.PROVIDER: getattr(module, module.SYNC_MONITOR_CLS)
+        for module in module_registry.values()
+        if hasattr(module, "SYNC_MONITOR_CLS")
+    }
+
+    for provider_name, _ in providers.items():
+        if provider_name not in monitor_classes:
+            monitor_classes[provider_name] = monitor_classes["generic"]
+
+    return monitor_classes
```

Comment:
This was extracted from `SyncService` intializer so I can use this from the script. This code is itself terribly dynamic but sync-engine was built to be extended by installing external modules like plugins, something that we don't do ourselves but I don't want to change it right n ow.